rotkehlchen/db/upgrades/v46_v47.py

```diff
@@ -32,35 +31,33 @@ def _add_new_locations(write_cursor: 'DBCursor') -> None:
     def _clean_cache(write_cursor: 'DBCursor') -> None:
         write_cursor.execute("DELETE FROM key_value_cache WHERE name LIKE 'extrainternaltx_%'")
 
-    @progress_step(description='Updating ERC721 token identifiers.')
-    def _update_erc721_token_identifiers(write_cursor: 'DBCursor') -> None:
-        """Update erc721 token identifiers to include the token id
-        for all evm events that have the token id in extra_data.
+    @progress_step(description='Remove unneeded nft collection assets.')
+    def _remove_nft_collection_assets(write_cursor: 'DBCursor') -> None:
+        """Remove erc721 assets that have no collectible id, and also any erc20 assets
+        with the same address that were incorrectly added.
         """
-        write_cursor = write_cursor.execute("SELECT identifier, asset, extra_data FROM history_events WHERE entry_type = 2 AND extra_data != ''")  # noqa: E501
-        assets_updates = []
-        extra_data_updates = []
-        for entry in write_cursor:
-            asset_identifier = entry[1]
-            try:
-                extra_data: dict = json.loads(entry[2])
-            except json.JSONDecodeError as e:
-                log.error(f'Failed to parse {entry[2]} as JSON due to {e}')
-                continue
-
-            if (token_id := extra_data.get('token_id')) is not None:
-                assets_updates.append((f'{asset_identifier}/{token_id}', asset_identifier))
-                # remove the token id and name from extra_data to avoid storing redundant data,
-                extra_data.pop('token_id', '')
-                extra_data.pop('token_name', '')
-                extra_data_updates.append((json.dumps(extra_data), entry[0]))
-
-        if len(assets_updates) > 0:
-            assets_query = 'UPDATE assets SET identifier=? WHERE identifier=?'
-            with GlobalDBHandler().conn.write_ctx() as global_db_write_cursor:
-                global_db_write_cursor.executemany(assets_query, assets_updates)
-
-            write_cursor.executemany(assets_query, assets_updates)
-            write_cursor.executemany('UPDATE history_events SET extra_data=? WHERE identifier=?', extra_data_updates)  # noqa: E501
+        identifiers_to_remove = []
+        with GlobalDBHandler().conn.read_ctx() as global_db_cursor:
+            cursor = global_db_cursor.execute("SELECT identifier FROM assets WHERE identifier LIKE '%erc721%'")  # noqa: E501
+            for entry in cursor:
+                identifiers_to_remove.extend([
+                    entry[0],
+                    entry[0].replace('erc721', 'erc20'),
+                ])
+
+        if len(identifiers_to_remove) == 0:
+            return
+
+        placeholders = ','.join(['?'] * len(identifiers_to_remove))
+        with GlobalDBHandler().conn.write_ctx() as global_db_write_cursor:
```

Comment:
just a nitpick but you can store the conn using the walrus operator and avoid invoking the new method on the globaldb